gammapy/estimators/tests/test_energydependence.py

```diff
@@ -0,0 +1,201 @@
+    lon_0 = Parameter("lon_0", "5.6 deg")
+    lat_0 = Parameter("lat_0", "0.2 deg", min=-90, max=90)
+
+    sigma_1TeV = Parameter("sigma_1TeV", "0.3 deg", min=0)
+    sigma_10TeV = Parameter("sigma_10TeV", "0.15 deg", min=0)
+
+    SpatialModel.frame = "galactic"
+
+    @staticmethod
+    def evaluate(lon, lat, energy, lon_0, lat_0, sigma_1TeV, sigma_10TeV):
+        sep = angular_separation(lon, lat, lon_0, lat_0)
+
+        # Compute sigma for the given energy using linear interpolation in log energy
+        sigma_nodes = u.Quantity([sigma_1TeV, sigma_10TeV])
+        energy_nodes = [1, 10] * u.TeV
+        log_s = np.log(sigma_nodes.to("deg").value)
+        log_en = np.log(energy_nodes.to("TeV").value)
+        log_e = np.log(energy.to("TeV").value)
+        sigma = np.exp(np.interp(log_e, log_en, log_s)) * u.deg
+
+        exponent = -0.5 * (sep / sigma) ** 2
+        norm = 1 / (2 * np.pi * sigma**2)
+
+        return norm * np.exp(exponent)
+
+    @property
+    def evaluation_radius(self):
+        """Evaluation radius (`~astropy.coordinates.Angle`)."""
+        return 2 * np.max([self.sigma_1TeV.value, self.sigma_10TeV.value]) * u.deg
+
+
+def create_dataset():
+    source_lat = 0.2
+    source_lon = 5.6
+    spatial_model = MyCustomGaussianModel()
+
+    spectral_model = PowerLawSpectralModel(
+        index=3, amplitude="1e-11 cm-2 s-1 TeV-1", reference="1 TeV"
+    )
+    model = SkyModel(
+        spatial_model=spatial_model, spectral_model=spectral_model, name="model1"
+    )
+
+    irfs = load_irf_dict_from_file(
+        "$GAMMAPY_DATA/cta-1dc/caldb/data/cta/1dc/bcf/South_z20_50h/irf_file.fits"
+    )
+    livetime = 10.0 * u.hr
+    location = observatory_locations["cta_south"]
+    # TO DO: update to remove 'mode' in 1.3
+    pointing = FixedPointingInfo(
+        mode=PointingMode.POINTING,
+        fixed_icrs=SkyCoord(source_lon, source_lat, unit="deg", frame="galactic").icrs,
+    )
+
+    obs = Observation.create(
+        pointing=pointing,
+        livetime=livetime,
+        irfs=irfs,
+        location=location,
+    )
+
+    # Define map geometry for binned simulation
+    energy_reco = MapAxis.from_edges(
+        np.logspace(-1.0, 2, 20), unit="TeV", name="energy", interp="log"
+    )
+    geom = WcsGeom.create(
+        skydir=(source_lon, source_lat),
+        binsz=0.02,
+        width=5 * u.deg,
+        frame="galactic",
+        axes=[energy_reco],
+    )
+
+    # Make the MapDataset
+    empty = MapDataset.create(geom, name="dataset-simu")
+    maker = MapDatasetMaker(selection=["exposure", "background", "psf", "edisp"])
+    maker_safe_mask = SafeMaskMaker(methods=["offset-max"], offset_max=2.0 * u.deg)
+    dataset = maker.run(empty, obs)
+    dataset = maker_safe_mask.run(dataset, obs)
+
+    # Add the model on the dataset and Poission fluctuate
+    dataset.models = model
+    dataset.fake(random_state=42)
+
+    return dataset
+
+
+source_pos = SkyCoord(5.58, 0.2, unit="deg", frame="galactic")
```

Comment:
You should not leave test code outside functions or class.

You have two options:
- create a fixture/function to create the model and then in each test function call the `create_model` and `create_dataset` functions
- Create a general Test class that will perform this code on init and then have test methods for the various things required.